golem-worker-service-base/src/gateway_api_definition/http/api_oas_convert.rs

```diff
@@ -0,0 +1,1097 @@
+    let media_type = openapiv3::MediaType {
+        schema: Some(openapiv3::ReferenceOr::Item(body_schema)),
+        ..Default::default()
+    };
+
+    let mut content = indexmap::IndexMap::new();
+    content.insert("application/json".to_string(), media_type);
+
+    // If we have response_mapping_input, body is always required
+    let required = true;
+
+    openapiv3::RequestBody {
+        description: Some("Request payload".to_string()),
+        content,
+        required,
+        extensions: Default::default(),
+    }
+}
+
+// Helper function: Determines request body schema
+fn determine_request_body_schema(response_mapping_input: &RibInputTypeInfo) -> openapiv3::Schema {
+    // Check request key and then look for body field within the request
+    if let Some(AnalysedType::Record(request_record)) = response_mapping_input.types.get("request")
+    {
+        if let Some(body_field) = request_record.fields.iter().find(|f| f.name == "body") {
+            return create_schema_from_analysed_type(&body_field.typ);
+        }
+    }
+
+    // Fallback to default object schema if no body is found
+    create_default_object_schema()
+}
+
+// Helper function: Gets default status code based on method
+fn get_default_status_code(method: &MethodPattern) -> u16 {
+    match method {
+        MethodPattern::Get => 200,
+        MethodPattern::Post => 201,
+        MethodPattern::Put => 204,
+        MethodPattern::Delete => 204,
+        MethodPattern::Options => 200,
+        MethodPattern::Head => 200,
+        MethodPattern::Patch => 200,
+        MethodPattern::Trace => 200,
+        _ => 200,
+    }
+}
+
+// Helper function: Create response
+fn create_response(status_code: u16, route: &RouteResponseData) -> openapiv3::Response {
+    let mut response = openapiv3::Response {
+        description: get_status_description(status_code),
+        ..Default::default()
+    };
+
+    // Only add content for non-204 responses
+    if status_code != 204 {
+        let mut content = indexmap::IndexMap::new();
+        let media = openapiv3::MediaType {
+            schema: Some(openapiv3::ReferenceOr::Item(determine_response_schema(
+                route,
+            ))),
+            ..Default::default()
+        };
+        content.insert("application/json".to_string(), media);
+        response.content = content;
+    }
+
+    response
+}
+
+// Helper function: Gets status code description
+fn get_status_description(status_code: u16) -> String {
+    match status_code {
+        204 => "No Content".to_string(),
+        201 => "Created".to_string(),
+        200 => "OK".to_string(),
+        400 => "Bad Request".to_string(),
+        404 => "Not Found".to_string(),
+        500 => "Internal Server Error".to_string(),
+        _ => "Response".to_string(),
+    }
+}
+
+// Helper function: Determines response schema
+fn determine_response_schema(route: &RouteResponseData) -> openapiv3::Schema {
```

Comment:
I think better to use `CompiledRoute` instead of `RouteResponseData`./
I think you will end up having to use it because of [this](https://github.com/golemcloud/golem/pull/1454/files#r2093467076)